python/ray/tests/test_output.py

```diff
@@ -190,9 +189,9 @@ def foo(out_str, err_str):
     out_str = proc.stdout.read().decode("ascii")
     err_str = proc.stderr.read().decode("ascii")
 
-    assert out_str.endswith("abc\n"), out_str
-    assert "(foo pid=" in out_str, out_str
-    assert err_str.split("\n")[-2].endswith("def")
+    assert out_str.rstrip("\r\n").endswith("abc"), out_str
```

Comment:
We should probably use https://docs.python.org/3/library/stdtypes.html#str.splitlines for all of these things to be clearer, what do you think?

Reply:
Sounds like a better option to me. Let me try it.

Reply:
Done.